packages/frontend-app-api/src/wiring/parameters.ts

```diff
@@ -40,93 +49,210 @@ export function readAppExtensionParameters(
+    throw new Error(errorMsg('must be a string or an object'));
+  }
+  const keys = Object.keys(arrayEntry);
+  if (keys.length !== 1) {
+    const joinedKeys = keys.length ? `'${keys.join("', '")}'` : 'none';
+    throw new Error(errorMsg(`must have exactly one key, got ${joinedKeys}`));
+  }
 
-    const key = keys[0];
-    let value = arrayEntry[key];
+  const key = keys[0];
+  const value = arrayEntry[key];
 
-    let at = key.includes('/') ? key : undefined;
-    const id = at ? generateIndex() : key;
-    if (typeof value === 'boolean') {
-      if (at) {
-        throw new Error(
-          errorMsg('cannot be applied to an instance input', key),
-        );
-      }
-      value = { disabled: !value };
-    } else if (typeof value === 'string') {
-      value = { extension: value };
-    } else if (
-      typeof value !== 'object' ||
-      value === null ||
-      Array.isArray(value)
-    ) {
-      throw new Error(errorMsg('must be an object', key));
+  // Example YAML:
+  // - catalog.page.cicd: false
+  if (typeof value === 'boolean') {
+    if (key.includes('/')) {
+      const suggestion = key.split('/')[0];
+      throw new Error(
+        errorMsg(
+          `cannot target an extension instance input (key cannot contain slashes; did you mean '${suggestion}'?)`,
+          key,
+        ),
+      );
     }
 
-    const atRef = value.at;
-    if (atRef !== undefined && typeof atRef !== 'string') {
-      throw new Error(errorMsg('must be a string', key, 'at'));
-    } else if (atRef !== undefined) {
-      if (at) {
-        throw new Error(
-          errorMsg(
-            `must not specify 'at' when using attachment shorthand form`,
-            key,
-          ),
-        );
-      }
-      at = atRef;
-    }
-    const disabled = value.disabled;
-    if (disabled !== undefined && typeof disabled !== 'boolean') {
-      throw new Error(errorMsg('must be a boolean', key, 'disabled'));
-    }
-    const extensionRef = value.extension;
-    if (extensionRef !== undefined && typeof extensionRef !== 'string') {
-      throw new Error(errorMsg('must be a string', key, 'extension'));
+    return {
+      id: key,
+      disabled: !value,
+    };
+  }
+
+  // Example YAML:
+  // - core.router/routes: '@backstage/plugin-some-plugin#MyPage'
+  // - some-plugin.page: '@internal/frontend-customizations#MyModifiedPage'
+  if (typeof value === 'string') {
+    if (key.includes('/')) {
+      return {
+        id: generateExtensionId(),
+        at: key,
+        extension: resolveExtensionRef(value),
+      };
     }
-    const extension = extensionRef
-      ? resolveExtensionRef(extensionRef)
-      : undefined;
+
     return {
-      id,
-      at,
-      disabled,
-      extension,
-      config: value.config /* validate later */,
+      id: key,
+      extension: resolveExtensionRef(value),
     };
-  });
+  }
+
+  // The remaining case is the generic object. Example YAML:
+  // - core.router/routes:
+  //     id: redirects.catalog
```

Comment:
Let's not allow an ID here imo, if one wants to specify an ID they should use it as a key. This effectively just ends up being an alternate format where id and at are flipped

Reply:
Aye, a worthwhile simplification. Fixed